orca-mine/src/test/java/com/netflix/spinnaker/orca/mine/tasks/RegisterCanaryTaskTest.java

```diff
@@ -0,0 +1,211 @@
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.springframework.http.HttpStatus;
+import retrofit.RestAdapter;
+import retrofit.client.OkClient;
+import retrofit.client.Response;
+import retrofit.converter.JacksonConverter;
+import retrofit.mime.TypedString;
+
+public class RegisterCanaryTaskTest {
+
+  private static MineService mineService;
+
+  private static RegisterCanaryTask registerCanaryTask;
+
+  private static StageExecution deployCanaryStage;
+
+  private static ObjectMapper objectMapper = new ObjectMapper();
+
+  @RegisterExtension
+  static WireMockExtension wireMock =
+      WireMockExtension.newInstance().options(wireMockConfig().dynamicPort()).build();
+
+  @BeforeAll
+  static void setupOnce(WireMockRuntimeInfo wmRuntimeInfo) {
+    OkClient okClient = new OkClient();
+    RestAdapter.LogLevel retrofitLogLevel = RestAdapter.LogLevel.NONE;
+
+    mineService =
+        new RestAdapter.Builder()
+            .setEndpoint(wmRuntimeInfo.getHttpBaseUrl())
+            .setClient(okClient)
+            .setLogLevel(retrofitLogLevel)
+            .setLog(new RetrofitSlf4jLog(MineService.class))
+            .setConverter(new JacksonConverter(objectMapper))
+            .build()
+            .create(MineService.class);
+
+    registerCanaryTask = new RegisterCanaryTask();
+    registerCanaryTask.setMineService(mineService);
+  }
+
+  @BeforeEach
+  public void setup() {
+    var pipeline = PipelineExecutionImpl.newPipeline("foo");
+
+    var canaryStageId = UUID.randomUUID().toString();
+    var parentStageId = UUID.randomUUID().toString();
+
+    deployCanaryStage = getDeployCanaryStage(pipeline, canaryStageId);
+    deployCanaryStage.setParentStageId(parentStageId);
+    var monitorCanaryStage =
+        new StageExecutionImpl(pipeline, "monitorCanary", Collections.emptyMap());
+
+    pipeline.getStages().addAll(List.of(deployCanaryStage, monitorCanaryStage));
+  }
+
+  private static void simulateFault(String url, String body, HttpStatus httpStatus) {
+    wireMock.givenThat(
+        WireMock.post(urlPathEqualTo(url))
+            .willReturn(
+                aResponse()
+                    .withHeaders(HttpHeaders.noHeaders())
+                    .withStatus(httpStatus.value())
+                    .withBody(body)));
+  }
+
+  @Test
+  public void verifyRegisterCanaryThrowsHttpError() throws Exception {
+
+    Response response =
+        new Response(
+            "https://mine.service.com/registerCanary",
+            HttpStatus.NOT_ACCEPTABLE.value(),
+            HttpStatus.NOT_ACCEPTABLE.getReasonPhrase(),
+            List.of(),
+            new TypedString("canaryId"));
+    String errorResponseBody = objectMapper.writeValueAsString(response);
+
+    // simulate error HTTP 406
+    simulateFault("/registerCanary", errorResponseBody, HttpStatus.NOT_ACCEPTABLE);
+
+    // Format as per the error response thrown by register canary task.
+    errorResponseBody =
```

Comment:
This doesn't get used til lower down, so perhaps move it there?

Reply:
This formatting part is removed, because the expected order of the response details printed in the logs may not match with the actual order. 
So, the alternative used here to assert is "hasMessageContaining()".